tests/io-test.ts

```diff
@@ -110,9 +149,16 @@ describe('io', () => {
 
       writeTodoStorageFile(todoStorageFilePath, operations);
 
-      compactTodoStorageFile(tmp, buildReadOptions());
+      const { originalOperations, compactedOperations, compacted } = compactTodoStorageFile(tmp, buildReadOptions());
 
       expect(readTodoStorageFile(todoStorageFilePath)).toEqual(operations);
+
+      const todoContents = readFileSync(todoStorageFilePath, { encoding: 'utf8' });
+      expect(todoContents.endsWith(EOL)).toEqual(true)
+
+      expect(originalOperations).toEqual(operations)
+      expect(compactedOperations).toEqual(operations)
+      expect(compacted).toEqual(0)
```

Comment:
Along with checking the final EOL, the tests now also check the operations & counts returned by the compact function.